core-web/libs/sdk/experiments/src/lib/dot-experiments.ts

```diff
@@ -197,20 +260,60 @@ export class DotExperiments {
      *
      * @private
      */
-    private async initAnalyticsClient() {
+    private initAnalyticsClient() {
         try {
             this.analytics = jitsuClient({
                 key: this.config['api-key'],
                 tracking_host: this.config['server'],
                 log_level: this.config['debug'] ? DEBUG_LEVELS.DEBUG : DEBUG_LEVELS.WARN
             });
 
-            const { experiments } = await this.getDataForAnalytics();
+            const { experiments } = parseDataForAnalytics(this.indexDBData.experiments, location);
             this.analytics.set({ experiments });
 
             dotLogger(`Analytics client created successfully.`, this.getIsDebugActive());
         } catch (error) {
             throw Error(`Error creating analytics client, ${error}`);
         }
     }
+
+    /**
+     * Determines whether analytics should be checked.
+     *
+     * @private
+     * @returns {Promise<boolean>} A boolean value indicating whether analytics should be checked.
+     */
+    private shouldCheckAnalytics(): boolean {
+        // If the user close the tab, reload the data
+        if (!checkFlagExperimentAlreadyChecked()) {
+            this.databaseHandler.clearData();
+
+            return true;
+        }
+
+        if (!this.indexDBData) {
+            return true;
+        }
+
+        if (this.indexDBData && checkInvalidateDataChecked(this.indexDBData.created)) {
+            return true;
+        }
+
+        dotLogger(`Not should Check Analytics by now...`, this.getIsDebugActive());
+
+        return false;
+    }
+
+    /**
+     * Retrieves persisted data from the database.
+     *
+     * @private
+     * @returns {Promise<void>} A promise that resolves with no value.
+     */
+    private async getPersistedData(): Promise<void> {
```

Comment:
if is `get` it should return.

Reply:
the next PR, I used it as the name says, I returned the data from the persisted data. I will leave this one as now.